app/lib/task/backend.dart

```diff
@@ -1169,6 +1161,41 @@ String? _extractBearerToken(shelf.Request request) {
   return parts.last.trim();
 }
 
+PackageVersionStateInfo _extractAndVerifyVersionState(
+  String package,
+  String version,
+  PackageState? state,
+  shelf.Request request,
+) {
```

Comment:
It's not clear what this method does.

Indeed if it's responsibility is to do authorization, that seems like something very important to highlight.

Perhaps rename it:
```suggestion
/// Authorize a worker callback for [package] / [version].
///
/// Returns the [PackageVersionStateInfo] that the worker is authenticated for.
/// Or throw [ResponseException] if authorization is not possible.
PackageVersionStateInfo _authorizeWorkerCallback(
  shelf.Request request,
  PackageState state,
  String package,
  String version,
) {
```

Avoid making `PackageState` nullable, if it's missing, that's not an authorization issue.